transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/persistence/SQLiteEventStore.java

```diff
@@ -352,7 +352,27 @@ public void recordNextCallTime(TransportContext transportContext, long timestamp
   public int cleanUp() {
     long oneWeekAgo = wallClock.getTime() - config.getEventCleanUpAge();
     return inTransaction(
-        db -> db.delete("events", "timestamp_ms < ?", new String[] {String.valueOf(oneWeekAgo)}));
+        db -> {
+          String query =
+              "SELECT COUNT(*), transport_name FROM events "
+                  + "INNER JOIN transport_contexts ON (events.context_id=transport_contexts._id) "
+                  + "WHERE timestamp_ms < ? "
+                  + "GROUP BY transport_name";
```

Comment:
Why do we need a join here? can't we just remove it?

In general I think there is risk that this query could be very expensive. Can you run some `explain` queries against the database and provide the output?

Reply:
Originally, i was planning to check the transportContext if we want to log or not, in that case, we will use 

```
            SELECT COUNT(*), transport_name, backend_name, priority, extras FROM events 
            INNER JOIN transport_contexts ON (events.context_id=transport_contexts._id) 
            WHERE timestamp_ms < ? 
            GROUP BY transport_name, backend_name, priority, extras
```
in that case, we will not need to join those two tables, will remove it.